integration-cli/docker_cli_run_windows_test.go

```diff
@@ -0,0 +1,52 @@
+// +build windows
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"strings"
+
+	winio "github.com/Microsoft/go-winio"
+	"github.com/go-check/check"
+)
+
+func (s *DockerSuite) TestBindNamedPipeToGuest(c *check.C) {
+	testRequires(c, SameHostDaemon, DaemonIsWindowsBuild(16210)) // Named pipe support was added in RS3
+
+	// Create a host pipe to map into the container
+	hostPipeName := fmt.Sprintf(`\\.\pipe\docker-cli-test-pipe-%x`, rand.Uint64())
+	pc := &winio.PipeConfig{
+		SecurityDescriptor: "D:P(A;;GA;;;AU)", // Allow all users access to the pipe
+	}
+	l, err := winio.ListenPipe(hostPipeName, pc)
+	if err != nil {
+		c.Fatal(err)
+	}
+	defer l.Close()
+
+	var b []byte
+	ch := make(chan error)
+	go func() {
+		conn, err := l.Accept()
+		if err == nil {
+			b, err = ioutil.ReadAll(conn)
+			conn.Close()
+		}
+		ch <- err
+	}()
+
+	containerPipeName := `\\.\pipe\docker-cli-test-pipe`
+	text := "hello from a pipe"
+	cmd := fmt.Sprintf("echo %s > %s", text, containerPipeName)
+	dockerCmd(c, "run", "-v", hostPipeName+":"+containerPipeName, testEnv.MinimalBaseImage(), "cmd", "/c", cmd)
```

Comment:
Yes, please convert the test to API test?
https://forums.mobyproject.org/t/evolution-of-testing-moby/38